### DISCUSSION TITLE
[Feature] Add support for nerdctl

### DISCUSSION THREAD

**Discussion by 89luca89** [89luca89/distrobox#1902] [Ideas]
We should work with both podman and nerdctl directly
if both are available, better use podman

**Comment by 89luca89** [comment]
keeping it in suspend for now

~~- missing `create` command~~
~~- missing `--ipc`~~
~~- missing `--type` in `nerdctl inspect`~~

- missing `--userns`

**Comment by dawidd6** [comment]
`--ipc` flag is supported: https://github.com/containerd/nerdctl/blob/660680b7ddfde1d38a66ec1c7f08f8d89ab92c68/cmd/nerdctl/run.go#L147

`--type` flag is also supported: https://github.com/containerd/nerdctl/blob/660680b7ddfde1d38a66ec1c7f08f8d89ab92c68/cmd/nerdctl/inspect.go#L55

Waiting for `--userns`.

**Comment by 89luca89** [comment]
> `--ipc` flag is supported: https://github.com/containerd/nerdctl/blob/660680b7ddfde1d38a66ec1c7f08f8d89ab92c68/cmd/nerdctl/run.go#L147
> 
> `--type` flag is also supported: https://github.com/containerd/nerdctl/blob/660680b7ddfde1d38a66ec1c7f08f8d89ab92c68/cmd/nerdctl/inspect.go#L55
> 
> Waiting for `--userns`.

Yea I know I've done it :joy: 

But you're right, I need to update the comment above :+1:

**Comment by profetik-777** [comment]
An older one - looks like almost done or done? Should we close it? @89luca89

**Comment by 89luca89** [comment]
Still not compatible
It still needs support for `--userns keep-id` or something similar in order to have uid-user-inside == uid-user-outside

Haven't had time to work on this on the nerdctl side sadly

**Comment by matthew-salerno** [comment]
[Relevant nerdctl PR to watch]( 
https://github.com/containerd/nerdctl/pull/3941)

**Comment by AkihiroSuda** [comment]
> --userns keep-id

Is this a hard requirement?
Probably rootful mode does not need this?

**Comment by 89luca89** [comment]
> > --userns keep-id
> 
> Is this a hard requirement? Probably rootful mode does not need this?

Not strictly, it's needed for rootless containers, which is what distrobox would prefer (hence the default to podman)
As you mention, it is not needed for rootful mode, in fact docker does not need it

**Comment by madliani** [comment]
@89luca89, this PR has already been merged.
